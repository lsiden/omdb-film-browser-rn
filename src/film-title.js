import React from "react"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import { TouchableHighlight, Text } from "react-native"

import { viewFilmSummary, fetchFilmDetails } from "./actions"
import { TITLE_COLOR } from "./constants"

const textStyle = {
  lineHeight: 30,
  fontSize: 18,
}
export const filmTitle = ({ filmSummary, dispatchViewDetail }) => (
  <TouchableHighlight onPress={() => dispatchViewDetail(filmSummary)}>
    <Text style={textStyle}>{`${filmSummary.Title}, ${filmSummary.Year}`}</Text>
  </TouchableHighlight>
)

filmTitle.propTypes = {
  filmSummary: PropTypes.object.isRequired,
  dispatchViewDetail: PropTypes.func.isRequired,
}

export default connect(null, dispatch => ({
  dispatchViewDetail: filmSummary => {
    dispatch(viewFilmSummary(filmSummary))
    dispatch(fetchFilmDetails(filmSummary.imdbID))
  },
}))(filmTitle)
