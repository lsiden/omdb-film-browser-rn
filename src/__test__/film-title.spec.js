import React from "react"
import renderer from "react-native-test-utils"

import { filmTitle as FilmTitle } from "film-title"

const films = require("./__fixture__/films.json").Search
let dispatchViewDetail

const defaultProps = () => {
  dispatchViewDetail = jest.fn()
  return {
    filmSummary: films[0],
    dispatchViewDetail,
  }
}

jest.mock("TouchableHighlight", () => {
  const mockComponent = require("jest-react-native")
  return mockComponent("TouchableHighlight")
})

const createWrapper = (props = {}) =>
  renderer(<FilmTitle {...{ ...defaultProps(), ...props }} />)

it("renders", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

test("click on title invokes dispatchViewDetail()", () => {
  const wrapper = createWrapper()
  wrapper.props.onPress()
  expect(dispatchViewDetail).toHaveBeenCalled()
})
