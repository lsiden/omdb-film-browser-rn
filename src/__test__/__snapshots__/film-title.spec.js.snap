// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  addStyle={
    Object {
      "color": "#210d87",
    }
  }
  delayPressOut={100}
  onPress={[Function]}
  underlayColor="black"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    The Hobbit: The Battle of the Five Armies, 2014
  </Text>
</TouchableHighlight>
`;
